src/query/sql/src/planner/semantic/type_check.rs

```diff
@@ -200,8 +199,19 @@ impl<'a> TypeChecker<'a> {
                         )
                     }
                     NameResolutionResult::InternalColumn(column) => {
-                        let data_type = column.internal_column.data_type();
-                        (BoundInternalColumnRef { column }.into(), data_type)
+                        // add internal column binding into `BindContext`
+                        let column = self
+                            .bind_context
+                            .add_internal_column_binding(&column, self.metadata.clone());
+                        let data_type = *column.data_type.clone();
+                        (
+                            BoundColumnRef {
+                                span: *span,
+                                column,
+                            }
+                            .into(),
+                            data_type,
+                        )
```

Comment:
@lichuang The InternalColumnBinding will be added to the bind_context and converted to a ColumnBinding once it is resolved. Therefore, InternalColumn can be removed from ScalarExpr.